pkg/bindings/connection.go

```diff
@@ -108,23 +111,76 @@ func NewConnectionWithIdentity(ctx context.Context, uri string, identity string,
 	var connection Connection
 	switch _url.Scheme {
 	case "ssh":
+		userinfo := _url.User
+		if _url.User == nil {
+			u, err := user.Current()
+			if err != nil {
+				return nil, fmt.Errorf("current user could not be determined: %w", err)
+			}
+			userinfo = url.User(u.Username)
+		}
 		port := 22
 		if _url.Port() != "" {
 			port, err = strconv.Atoi(_url.Port())
 			if err != nil {
 				return nil, err
 			}
 		}
+		// ssh_config
+		alias := _url.Hostname()
+		logrus.Debugf("Alias: %s", alias)
+		cfg := ssh_config.DefaultUserSettings
+		if val := cfg.Get(alias, "User"); val != "" {
+			userinfo = url.User(val)
+			logrus.Debugf("User: %s", val)
+		}
+		if val := cfg.Get(alias, "Hostname"); val != "" {
+			uri = val
+			logrus.Debugf("Hostname: %s", val)
+		}
+		if val := cfg.Get(alias, "Port"); val != "" {
+			if val != ssh_config.Default("Port") {
+				port, err = strconv.Atoi(val)
+				if err != nil {
+					return nil, fmt.Errorf("port is not an int: %s: %w", val, err)
+				}
+				logrus.Debugf("Port: %s", val)
+			}
+		}
+		if val := cfg.Get(alias, "IdentityFile"); val != "" {
+			if val != ssh_config.Default("IdentityFile") {
+				identity = strings.Trim(val, "\"")
+				logrus.Debugf("IdentityFile: %s", val)
+			}
+		}
 		conn, err := ssh.Dial(&ssh.ConnectionDialOptions{
 			Host:                        uri,
 			Identity:                    identity,
-			User:                        _url.User,
+			User:                        userinfo,
 			Port:                        port,
 			InsecureIsMachineConnection: machine,
 		}, "golang")
 		if err != nil {
 			return nil, newConnectError(err)
 		}
+		// socket path
+		if _url.Path == "" {
+			session, err := conn.NewSession()
+			if err != nil {
+				return nil, err
+			}
+			defer session.Close()
+
+			var b bytes.Buffer
+			session.Stdout = &b
+			if err := session.Run(
+				"podman info --format '{{.Host.RemoteSocket.Path}}'"); err != nil {
+				return nil, err
+			}
```

Comment:
This is not a good idea for performance reasons, podman info is slow...
Also podman info has no actual knowledge of the actual API socket path and just guess the /run/podman/podman.sock or /run/user/xxx/podman/podman.sock as rootless so there is no reason to do that as you might as well guess on the client. Of course you would need to get the uid on the server to do this.

I mean alone the extra ssh connection is already slow but I think there is no need to avoid that if we need the uid on the server so maybe the podman info at this point is not to bad either?
Given that is is properly not a good idea to recommend leaving it the socket path if users care about performance.

Reply:
I think it was more about compatibility/simplicity, and less about performance?

There is already the "system connection", that should be the preferred option...

Reply:
I don't think it can hardcode the `/run/user/$UID` either, need to query for $XDG_RUNTIME_DIR.

Most systems with systemd will use the default, but other systems might use another directory...

Reply:
Yeah I looked at this was like podman info isn't really great for this but thinking about this again given there is no way around opening an extra ssh connection for it so doing the podman info call should not be that big of a deal either.

Reply:
Eventually it might use the containers/common functionality for this (and the "SSHMode").

Reply:
https://github.com/containers/common/tree/main/pkg/ssh